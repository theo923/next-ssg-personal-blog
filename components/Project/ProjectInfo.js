
import React from 'react';



const ProjectInfo = () => {

    const ProjectGenerator = (link, imgUrl, heading, context) => {
        return(
            <div class="w-full p-6 mx-auto md:w-1/2 xl:w-1/3">
                <a href={link} target='_blank'>
                    <div class="p-6 my-4 lg:p-8 cursor-pointer rounded-xl bg-gray-100 hover:bg-blue-200 shadow-2xl transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110">
                        <img class="object-cover object-center h-64 w-full mb-8 rounded-xl" src={imgUrl} alt={heading}/>
                        <h1 class="font-spartan mx-auto mb-8 text-2xl font-semibold leading-normal tracking-tighter text-black title-font">{heading}</h1>
                        <p class=" font-hindSiliguri mx-auto text-lg leading-loose text-blueGray-700 h-64 ">{context}</p>
                    </div>
                </a>
            </div>
        )
    }

    return (
        <div>
            <section class="text-blueGray-700">
                <div class="container items-center px-5 py-64 mx-auto h-full lg:px-16">
                    <div class="flex flex-wrap mb-12 text-left">
                        {ProjectGenerator('https://theo923.com/', './image/Blog.png', 'My Personal Blog', 'Blog about my onging coding journey, my life, my works. Sometimes I will share my solution in dealing with different problems, sometimes I will share anythings that meaningful to me, sometimes I will share my happiness if I have accomplished some achievements...')}
                        {ProjectGenerator('https://react-shopping-mall.netlify.app/', './image/ShoppingMall.png', 'React Shopping Mall', 'Online Shopping Mall Website DEMO (PWA App) that using React JS & Typescript as Frontend, Express JS as Server, PostgreSQL as Database. It can announce on new products, perform checkout function on shopping cart, retreive the tracking information.')}
                        {ProjectGenerator('https://hub.docker.com/u/theo9234/', './image/ShoppingMallDockerImage.png', 'React Shopping Mall Docker', 'Online Shopping Mall Website DEMO Docker Image that can be used by Dockhub / Unraid Docker')}
                    </div>
                </div>
            </section>
        </div>
    )
    
}

export default ProjectInfo;


