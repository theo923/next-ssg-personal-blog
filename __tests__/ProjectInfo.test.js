import { shallow } from 'enzyme';
import React from 'react';
import ProjectInfo from '../components/Project/ProjectInfo';

describe('ProjectInfo test', () => {

    const setup = (props={}, state=null) => {
        return shallow(<ProjectInfo {...props} />)
    }

    const findJSXByAttr = (name, wrapper) => {
        return wrapper.find(`[data-test="${name}"]`)
    }

    it("expect ProjectInfo component is rendered without crashing", () => {
        const wrapper = setup();   
    });

    it("expect component-project is rendered", () => {
        const wrapper = setup();   
        const projectInfo = findJSXByAttr("component-project", wrapper)
        expect(projectInfo.length).toBe(1);
    });
    
});


