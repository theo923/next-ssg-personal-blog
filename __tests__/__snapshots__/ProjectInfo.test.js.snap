// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectInfo test expect ProjectInfo component is rendered && match with the Snapshot 1`] = `
<div>
  <section
    className="text-blueGray-700"
  >
    <div
      className="container items-center px-5 py-64 mx-auto h-full lg:px-16"
    >
      <div
        className="flex flex-wrap mb-12 text-left"
      >
        <div
          className="w-full p-6 mx-auto md:w-1/2 2xl:w-1/3"
        >
          <a
            href="https://netfake.vercel.app/"
            target="_blank"
          >
            <div
              className="p-6 my-4 lg:p-8 cursor-pointer rounded-xl bg-gray-100 hover:bg-blue-200 shadow-2xl transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110"
            >
              <img
                alt="Netfake - A Netflix main page Clone"
                className="object-cover object-center h-72 w-full mb-8 rounded-xl"
                src="./image/Netfake.png"
              />
              <h1
                className="font-spartan mx-auto mb-8 text-2xl font-semibold leading-normal tracking-tighter text-black title-font"
              >
                Netfake - A Netflix main page Clone
              </h1>
              <p
                className=" font-hindSiliguri mx-auto text-lg leading-loose text-blueGray-700 h-64 "
              >
                A netflix clone created by Reactjs, Typescript, TailwindCSS, AXIOS, Jest, Enzyme, Test Driven Development, and more....
              </p>
            </div>
          </a>
        </div>
        <div
          className="w-full p-6 mx-auto md:w-1/2 2xl:w-1/3"
        >
          <a
            href="https://theo923.com/"
            target="_blank"
          >
            <div
              className="p-6 my-4 lg:p-8 cursor-pointer rounded-xl bg-gray-100 hover:bg-blue-200 shadow-2xl transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110"
            >
              <img
                alt="My Personal Blog"
                className="object-cover object-center h-72 w-full mb-8 rounded-xl"
                src="./image/Blog.png"
              />
              <h1
                className="font-spartan mx-auto mb-8 text-2xl font-semibold leading-normal tracking-tighter text-black title-font"
              >
                My Personal Blog
              </h1>
              <p
                className=" font-hindSiliguri mx-auto text-lg leading-loose text-blueGray-700 h-64 "
              >
                Blog about my onging coding journey, my life, my works. Sometimes I will share my solution in dealing with different problems, sometimes I will share anythings that meaningful to me, sometimes I will share my happiness if I have accomplished some achievements...
              </p>
            </div>
          </a>
        </div>
        <div
          className="w-full p-6 mx-auto md:w-1/2 2xl:w-1/3"
        >
          <a
            href="https://react-shopping-mall.netlify.app/"
            target="_blank"
          >
            <div
              className="p-6 my-4 lg:p-8 cursor-pointer rounded-xl bg-gray-100 hover:bg-blue-200 shadow-2xl transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110"
            >
              <img
                alt="React Shopping Mall"
                className="object-cover object-center h-72 w-full mb-8 rounded-xl"
                src="./image/ShoppingMall.png"
              />
              <h1
                className="font-spartan mx-auto mb-8 text-2xl font-semibold leading-normal tracking-tighter text-black title-font"
              >
                React Shopping Mall
              </h1>
              <p
                className=" font-hindSiliguri mx-auto text-lg leading-loose text-blueGray-700 h-64 "
              >
                Online Shopping Mall Website DEMO (PWA App) that using React JS & Typescript as Frontend, Express JS as Server, PostgreSQL as Database. It can announce on new products, perform checkout function on shopping cart, retreive the tracking information.
              </p>
            </div>
          </a>
        </div>
        <div
          className="w-full p-6 mx-auto md:w-1/2 2xl:w-1/3"
        >
          <a
            href="https://hub.docker.com/u/theo9234/"
            target="_blank"
          >
            <div
              className="p-6 my-4 lg:p-8 cursor-pointer rounded-xl bg-gray-100 hover:bg-blue-200 shadow-2xl transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110"
            >
              <img
                alt="React Shopping Mall Docker"
                className="object-cover object-center h-72 w-full mb-8 rounded-xl"
                src="./image/ShoppingMallDockerImage.png"
              />
              <h1
                className="font-spartan mx-auto mb-8 text-2xl font-semibold leading-normal tracking-tighter text-black title-font"
              >
                React Shopping Mall Docker
              </h1>
              <p
                className=" font-hindSiliguri mx-auto text-lg leading-loose text-blueGray-700 h-64 "
              >
                Online Shopping Mall Website DEMO Docker Image that can be used by Dockhub / Unraid Docker
              </p>
            </div>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
